src/Find-SDC-Repositories.js

```diff
@@ -0,0 +1,119 @@
+const graphql = require("graphql-request");
+const { zipWith } = require("lodash");
+const prompt = require("prompt");
+const fs = require("fs");
+const process = require("process");
+const mkdirp = require("mkdirp");
+
+const config = {
+  file: {
+    filePrefix: "SDC-Teams-Github-Repos-",
+    directoryPath: "outputs/SDCTeamGithubRepos/"
+  },
+  api: {
+    endpoint: "https://api.github.com/graphql",
+    options: {
+      headers: {
+        Authorization: null
+      }
+    },
+    query: `
+    query { 
+        organization(login:"ONSDigital") { 
+            teams(first: 100) {
+                nodes {
+                    name
+                    repositories(first: 100) {
+                        nodes {
+                            name
+                        }
+                        edges {
+                            permission
+                        }
+                    }
+                }
+            }
+        }
+    }`
+  }
+};
+
+prompt.start();
+
+prompt.get(["key"], function(err, result) {
+  config.api.options.headers.Authorization = "Bearer " + result.key;
+
+  if (
+    config.api.options.headers.Authorization !== null &&
+    config.api.options.headers.Authorization !== ""
```

Comment:
If you import `lodash` you can replace this with the following:

```
if (isNil(get(config, "api.options.headers.Authorization"))) {
...
```

Reply:
Rather than have everything in a massive `if` block, it might make sense to reverse the logic here and fail fast if the auth header is not set.

Probably better to fail fast.